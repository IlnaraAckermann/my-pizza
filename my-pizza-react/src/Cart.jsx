import { useContext } from "react";
import { currencyFormatter } from "./untils/formatters";
import { CartContext } from "./context/context";

export default function Cart({ checkout }) {
  const [cart, setCart] = useContext(CartContext);
  let total = 0;

  total = cart.reduce((acc, item) => acc + item.pizza.sizes[item.size], 0);

  return (
    <div className="cart">
      <h2>Cart</h2>
      <ul>
        {cart.map((item) => (
          <li key={item.id}>
            <span className="size">{item.size}</span>
            <span className="type"> - {item.pizza.name} - </span>
            <span className="price">{item.price}</span>
            <button
              onClick={() =>
                setCart((prev) => prev.filter((e) => e.id !== item.id))
              }
              className="remove-button"
            >
              remove
            </button>
          </li>
        ))}
      </ul>
      <h3>Total: {currencyFormatter(total)}</h3>
      <button onClick={checkout}>Checkout</button>
    </div>
  );
}
