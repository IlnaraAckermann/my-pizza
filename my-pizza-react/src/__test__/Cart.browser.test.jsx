import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import Cart from "../Cart";
import { CartContext } from "../context/context";

test("should render with nothing in the cart", async () => {
  const { asFragment } = render(
    <CartContext.Provider value={[[], () => {}]}>
      <Cart />
    </CartContext.Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test("should render with has itens in the cart", async () => {
  const { asFragment } = render(
    <CartContext.Provider
      value={[
        [
          {
            pizza: {
              name: "pepperoni",
              sizes: { M: 3.0, L: 5.0, S: 1.0 },
            },
            size: "M",
            price: "$3.00",
            id: Math.random(),
          },
        ],
        () => {},
      ]}
    >
      <Cart />
    </CartContext.Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
