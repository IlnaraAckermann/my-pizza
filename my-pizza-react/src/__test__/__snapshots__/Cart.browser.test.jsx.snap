// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render with has itens in the cart 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    <h2>
      Cart
    </h2>
    <ul>
      <li>
        <span
          class="size"
        >
          M
        </span>
        <span
          class="type"
        >
           - pepperoni - 
        </span>
        <span
          class="price"
        >
          $3.00
        </span>
        <button
          class="remove-button"
        >
          remove
        </button>
      </li>
    </ul>
    <h3>
      Total: $3.00
    </h3>
    <button>
      Checkout
    </button>
  </div>
</DocumentFragment>
`;

exports[`should render with nothing in the cart 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    <h2>
      Cart
    </h2>
    <ul />
    <h3>
      Total: $0.00
    </h3>
    <button>
      Checkout
    </button>
  </div>
</DocumentFragment>
`;
